.github/workflows/bowline.yml

```
@@ -0,0 +1,158 @@
+name: bowline CI
+
+on:
+  push:
+    branches: [ 'main', 'ci', 'ci-uc' ]
+  pull_request:
+    branches: [ 'main' ]
+  release:
+    types:
+      - created
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'
+        required: false
+        default: false
+
+env:
+  GITHUB_PR_NUMBER: ${{github.event.pull_request.number}}
+
+jobs:
+  build-deb-and-tests:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:  [ 'ubuntu-latest' ]
+
+    steps:
+    - name: Setup tmate session
+      uses: mxschmitt/action-tmate@v3
+      if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.debug_enabled }}
+      with:
+        detached: true
+    - name: env variable
+      run: |
+        echo "deb_version=0.0.0."$(date +%Y%m%d)"" >> $GITHUB_ENV
+        if [ ${{ github.event_name }} == "pull_request" ] ; then
+          echo "head_commit=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+        else
+          echo "head_commit=${{ github.sha }}" >> $GITHUB_ENV
+        fi
+    - uses: actions/checkout@v4
+    - uses: actions/cache@v4
+      id: cache
+      with:
+        path: |
+          cabal-cache
+        key: cache-${{ hashFiles('dnsext-bowline/dnsext-bowline.cabal') }}
+    - name: inject cache into docker
+      uses: reproducible-containers/buildkit-cache-dance@v3.1.0
+      with:
+        cache-map: |
+          {
+            "cabal-cache": "/root/.cabal"
+          }
+        skip-extraction: ${{ steps.cache.outputs.cache-hit }}
+
+    - name: build docker image (and binaries)
+      run: |
+        mkdir -p linux/pkg/opt/bowline/bin linux/pkg/opt/bowline/etc
+        cd build-docker
+        ./build.sh -x ${{ env.head_commit }} ${{ github.server_url }}/${{ github.repository }}
+        cd ..
+        docker run --rm -v /home:/home bowline:bookworm bash -c \
+            "cp /opt/bowline/bin/{bowline,dug,dump} `pwd`/"
+    - name: build deb package
+      run: |
+        mkdir -p pkg/work/bowline-${{ env.deb_version }}/bin
+        docker run --rm -i --name build-bowline \
+            -v /home:/home bowline:bookworm \
```

```suggestion
            -v /home:/home debian:bookworm-slim \
```

I believe it would be better to separate the dependencies used during deb-pkg building from the bowline docker-image.

---

??

I can use a plain debian image for bowline build,  but I don't understand if this makes any sense at all.
if you use the plain image to build, resulting product might be what current `build.sh` and co look like.

if you wish to focus more on testing docker builds, the you can do it.  but this is out of my intention to this patchset, to offer the ability to build/publish packages.

and what's the point of using `-slim` image in this context ?